compiler/aarch64/codegen/OMRTreeEvaluator.cpp

```diff
@@ -6400,10 +6400,74 @@ OMR::ARM64::TreeEvaluator::arraytranslateAndTestEvaluator(TR::Node *node, TR::Co
 
 TR::Register *
 OMR::ARM64::TreeEvaluator::arraytranslateEvaluator(TR::Node *node, TR::CodeGenerator *cg)
-	{
-	// TODO:ARM64: Enable TR::TreeEvaluator::arraytranslateEvaluator in compiler/aarch64/codegen/TreeEvaluatorTable.hpp when Implemented.
-	return OMR::ARM64::TreeEvaluator::unImpOpEvaluator(node, cg);
-	}
+   {
+   // tree looks as follows:
+   // arraytranslate
+   //    (0) input ptr
+   //    (1) output ptr
+   //    (2) translation table (dummy)
+   //    (3) stop character (terminal character, either 0xff00ff00 (ISO8859) or 0xff80ff80 (ASCII)
+   //    (4) input length (in elements)
+   //    (5) stopping char (dummy)
+   //
+   // Number of translated elements is returned
+
+   TR::Compilation *comp = cg->comp();
+
+   TR_ASSERT_FATAL(!node->isSourceByteArrayTranslate(), "Source is byte[] for arraytranslate");
+   TR_ASSERT_FATAL(node->isTargetByteArrayTranslate(), "Target is char[] for arraytranslate");
+   TR_ASSERT_FATAL(node->getChild(3)->getOpCodeValue() == TR::iconst && node->getChild(3)->getInt() == 0x0ff00ff00, "Non-ISO8859 stop character for arraytranslate");
```

Comment:
What is child(3) used for?  It does not appear to be used in this evaluator, so is this assert necessary?

Reply:
child(3) is used for the `arraytranslateTRTO` for ASCII translation.  I already have the code for that, and I am thinking of opening separate PRs for that after `arraytranslateTRTO255` (for ISO 8859-1 translation) gets merged.

Do you want to also review `arraytranslateTRTO` together with `arraytranslateTRTO255` in this PR?